doc/developer-documentation.md

```diff
@@ -17,105 +17,146 @@ In order to enable a new tree, we need to add an entry there.
     image: <docker-image-name>
-    params: <parameters such as arch, compiler, defconfig>
-    rules: <job-rules>
+    params:
+      arch: <architecture>
+      compiler: <compiler-name>
+      cross_compile: <compiler-prefix>
+      dtbs_check: <dtb-check-enabled>
+      defconfig: <defconfig-name>
+      fragments:
+       - <fragment-name>
+    rules:
+      min_version:
+        version: <kernel-version>
+        patchlevel: <kernel-patch_level>
+      tree:
+        - <tree-name>
+        - !<tree-name2>
+
+  <test-job-name>:
+    template: <template-name>
+    kind: <kind of job, either 'job' or 'test'>
+    params:
+      nfsroot: <rootfs-url>
+      collections: <name-of-suite>
+      job_timeout: <timeout>
     kcidb_test_suite: <kcidb-mapping>
+    rules:
+      min_version:
+        version: <kernel-version>
+        patchlevel: <kernel-patch-level>
+      tree:
+        - <tree-name>
+        - !<tree-name2>
 ```
-
 Here is the description of each field:
-- A `jinja2` template should be added to [`config/runtime`](https://github.com/kernelci/kernelci-pipeline/tree/main/config/runtime) directory.
-The template will be used to generate test definition.
-- `kind` field denotes type of the job. It should be `kbuild` for build jobs, `job` for a test suite, and simply `test` for a single test case.
-- `image` field specifies the Docker image used for building and running the test. Please note that this field is optional. Some jobs such as LAVA test jobs use image defined in the test definition template instead.
-- `params` includes parameters used for building the kernel (for `kbuild` job) or running the test, such as architecture, compiler, defconfig option, job timeout, etc.
-- `rules` will define job rules. If the test should be scheduled for specific kernel tree, branch, or version, these rules can be defined in this section.
-- `kcidb_test_suite` is a field used to map KernelCI test suite name with
-KCIDB test. Build jobs (with `kind: kbuild`) do not require this field.
-
+- **`template`**: A `jinja2` template should be added to the [`config/runtime`](https://github.com/kernelci/kernelci-pipeline/tree/main/config/runtime) directory. This template will be used to generate the test definition.
+- **`kind`**: The `kind` field specifies the type of job. It should be `kbuild` for build jobs, `job` for a test suite, and `test` for a single test case.
+- **`image`**: The `image` field specifies the Docker image used for building and running the test. This field is optional. For example, LAVA test jobs use an image defined in the test definition template instead.
+- **`params`**: The `params` field includes parameters for building the kernel (for `kbuild` jobs) or running the test. These parameters can include architecture, compiler, defconfig options, job timeout, etc.
+- **`rules`**: The `rules` field defines job rules. If a test should be scheduled for a specific kernel tree, branch, or version, these rules can be specified here.
```

Comment:
One last thing.
Please add a line here describing what rules mean with and without `!`.

Reply:
Sorry, but it seems a bit unclear to me.
Maybe something like "The rules with `!` will exclude that specific condition for job scheduling e.g. in the above example, the scheduler won't schedule a job if the event is received for Kernel tree `tree-name2`"?
Feel free to rephrase it if required.

Reply:
Changed:
"The rules prefixed with `!` exclude the specified condition from job scheduling. For example, in the given scenario, the scheduler does not schedule a job if an event is received for the kernel tree `tree-name2`."